tests/experimental_combination/attenuator_test.py

```diff
@@ -8,168 +11,214 @@
+#             "Route": "025A_DEC",
+#             "Measure": 5,
+#         },
+#         "features": [
+#             {
+#                 "properties": {
+#                     "core_details": {"description": ""},
+#                     "start_date": "2022-07-28T21:00:00Z",
+#                     "end_date": "2022-07-29T21:00:00Z",
+#                 },
+#                 "geometry": {},
+#             }
+#         ],
+#     }
+#     actual = attenuator.combine_geotab_with_wzdx(geotab_avl, wzdx)
+
+#     expected = {
+#         "properties": {
+#             "beginning_milepost": 0,
+#             "ending_milepost": 1,
+#         },
+#         "geometry": {
+#             "coordinates": [],
+#         },
+#     }
+#     print(attenuator.get_geometry_for_distance_ahead.call_args)
+#     attenuator.get_geometry_for_distance_ahead.assert_called_with(0, {}, 2, 5, 10)
+#     print(actual)
+#     assert actual == expected
+
+
+class MockGeospatialApi:
+    def __init__(
+        self, getCachedRequest=lambda x: None, setCachedRequest=lambda x, y: None
+    ):
+        self.getCachedRequest = getCachedRequest
+        self.setCachedRequest = setCachedRequest
+
+    def get_route_geometry_ahead(
+        self,
+        routeId,
+        startMeasure,
+        heading,
+        distanceAhead,
+        compressed=False,
+        routeDetails=None,
+        mmin=None,
+        mmax=None,
+    ):
+        return {"coordinates": "a", "start_measure": "b", "end_measure": "c"}
+
+
+@patch.object(
+    cdot_geospatial_api,
+    "GeospatialApi",
+    side_effect=lambda: MockGeospatialApi(),
+)
 def test_get_geometry_for_distance_ahead(cdot_patch):
     # cdot_geospatial_api.get_route_geometry_ahead = MagicMock(
     #     return_value={'coordinates': 'a', 'start_measure': 'b', 'end_measure': 'c'})
     actual = attenuator.get_geometry_for_distance_ahead(
-        0, {'Route': 0, 'Measure': 0}, 0, 0, 0)
-    expected = ('a', 'b', 'c')
+        0, {"Route": 0, "Measure": 0}, 0, 0, 0
+    )
+    expected = ("a", "b", "c")
     assert actual == expected
 
 
 def test_get_distance_ahead_normal():
-    actual = attenuator.get_distance_ahead_miles(25, 30*60)
+    actual = attenuator.get_distance_ahead_miles(25, 30 * 60)
     assert actual == 12.5
 
 
 def test_get_distance_ahead_normal_2():
-    actual = attenuator.get_distance_ahead_miles(10, 30*60)
+    actual = attenuator.get_distance_ahead_miles(10, 30 * 60)
     assert actual == 5
 
 
 def test_get_distance_ahead_default():
-    actual = attenuator.get_distance_ahead_miles(0, 30*60)
+    actual = attenuator.get_distance_ahead_miles(0, 30 * 60)
     assert actual == 2.5
 
 
-def test_main():
-    outputPath = './tests/data/output/wzdx_attenuator_combined.json'
-    try:
-        os.remove(outputPath)
-    except Exception:
-        pass
-    attenuator.main(outputPath=outputPath)
-    assert os.path.isfile(outputPath)
-    assert len(json.loads(open(outputPath).read())) == 1
+# def test_main():
```

Comment:
Invalid test or just failing? Not sure why it's commented.

Reply:
Same as above!